arangod/VocBase/Methods/Upgrade.cpp

```diff
@@ -305,8 +305,8 @@ UpgradeResult methods::Upgrade::runTasks(TRI_vocbase_t& vocbase, VersionResult&
     // check that the database occurs in the database list
     if (!(t.databaseFlags & dbFlag)) {
       // special optimization: for local server and new database,
-      // an upgrade-only task can be viewed as executed.
-      if (isLocal && dbFlag == DATABASE_INIT && t.databaseFlags == DATABASE_UPGRADE) {
+      // an one-shot task can be viewed as executed.
+      if (isLocal && dbFlag == DATABASE_INIT && (t.databaseFlags & DATABASE_ONLY_ONCE)) {
```

Comment:
where is `DATABASE_UPGRADE` handled then?

Reply:
dbFlag is set to DATABASE_UPGRADE if version check will report UPGRADE_NEEDED status see Upgrade.cpp:169 that databaseFlags for task will pass check on Upgrade.cpp:306 and task will be executed